ksql-rest-app/src/main/java/io/confluent/ksql/rest/util/KsqlUncaughtExceptionHandler.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.rest.util;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import io.confluent.ksql.rest.server.KsqlServerMain;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class KsqlUncaughtExceptionHandler implements Thread.UncaughtExceptionHandler {
+
+  private static final Logger log = LoggerFactory.getLogger(KsqlServerMain.class);
+
+  @SuppressFBWarnings
+  public void uncaughtException(final Thread t, final Throwable e) {
+    log.error("Unhandled exception caught in thread {}, error: {}", t.getName(), e.getStackTrace());
+    System.exit(-1);
```

Comment:
Couple things:

Can we also write the exception to stderr here?

We should call `org.apache.log4j.LogManager.shutdown()` to ensure that all the buffered logs (including the one above) get flushed. Ideally this would get passed in the constructor as a `Runnable`:
```
class KsqlUncaughtExceptionHandler implements UncaughtExceptionHandler {
    private final Runnable flusher;

    public void KsqlUncaughtExceptionHandler(final Runnable flusher) {
        this.flusher = flusher;
    }
}
...
new KsqlUncaughtExceptionHandler(LogManager::flush);
```

Reply:
I'll add it, but is there a reason why we don't call LogManager.shutdown() currently when the server exits? I don't see any other uses of it.

Reply:
We should definitely call it when the server exits cleanly as well. It's more critical here because we might lose the exact information we need to help debug the issue that made the thread crash.